OpenRA.Mods.RA/Traits/TargetableSubmarine.cs

```diff
@@ -34,7 +34,7 @@ public override string[] TargetTypes
 		{
 			get
 			{
-				return cloak.Cloaked ? info.CloakedTargetTypes : info.TargetTypes;
+				return IsTraitDisabled ? new string[] { } : cloak.Cloaked ? info.CloakedTargetTypes : info.TargetTypes;
```

Comment:
Bonus points if you extract that empty array into a static readonly variable instead of creating a new object each time.